mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+        MLFlowTracker._log_run(context=context, run=run)
+
+    @classmethod
+    def import_run(
+        cls,
+        project: MlrunProject,
+        pointer: str,
+        function_name: str = None,
+        handler: str = None,
+    ) -> RunObject:
+        """
+        Import a previous MLFlow experiment run to MLRun.
+
+        :param project:       The MLRun project to import the run to.
+        :param pointer:       The MLFlow `run_id` to import.
+        :param function_name: The MLRun function to assign this run to.
+        :param handler:       The handler for MLRun's RunObject
+
+        :return: The newly imported RunObject.
+        """
+        # Validate function name was given:
+        if function_name is None:
+            raise ValueError(
+                "For importing a MLFlow experiment run, a MLRun function name must be provided."
+            )
+
+        # Get the MLFlow run object:
+        run = mlflow.search_runs(
+            search_all_experiments=True,
+            filter_string=f"attributes.run_id = '{pointer}'",
+            output_format="list",
+        )
+        if not run:
+            raise ValueError(
+                "The provided run id was not found in the set MLFlow registry. Try to set the registry using "
+                "MLFlow's environment variables."
+            )
+        run = run[0]  # We are using a run id, so only one will be returned in the list.
+        # Create a run spec and metadata for creating the run object to hold the mlflow run:
+        run_spec = RunSpec(function=function_name, handler=handler)
+        run_metadata = RunMetadata(
+            uid=run.info.run_uuid, name=run.info.run_name, project=project.name
+        )
+        run_object = RunObject(spec=run_spec, metadata=run_metadata)
+        # Create a context from the run object:
+        ctx = mlrun.get_or_create_ctx(
+            name=run.info.run_name,
+            spec=run_object,
+        )
+        # Store the run in the MLRun DB, then import the MLFlow data to it:
+        ctx.store_run()
+        cls._log_run(context=ctx, run=run, is_offline=True)
+
+        # Create a rundb in order to update the run's state as completed (can't be done using context)
+        rundb = mlrun.get_run_db()
+        project = run_object.metadata.project
+        uid = run_object.metadata.uid
+        updates = {
+            "status.last_update": now_date().isoformat(),
+            "status.state": "completed",
+        }
+        run_object.status.state = "completed"
+        rundb.update_run(updates, uid, project)
+
+        # Print a summary message after importing the run:
+        # RuntimeMock needed for mocking runtime parameter in `_log_track_results`
+        class RuntimeMock:
+            is_child = False
+
+        result = ctx.to_dict()
+        run_object = RunObject.from_dict(result)
+        ClientBaseLauncher._log_track_results(
+            runtime=RuntimeMock, result=result, run=run_object
+        )
```

Comment:
id change `_log_track_results` signature to include "log_commands" and make callee use the `is_child` to determine that.

Reply:
can you please elaborate?

Reply:
instead of creating `RuntimeMock` with `is_child=False`, id change `_log_track_results` signature to take a variable other thn runtime to understand whether it needs to prent the relevant information of runtime who's not a child